File: everyshowing/src/main/java/com/wghcwc/everyshowing/view/LoadingDefaultView.java
package com.wghcwc.everyshowing.view;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.animation.Animation;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.wghcwc.everyshowing.R;
import com.wghcwc.everyshowing.loadstyle.BaseLoadingStyle;
import com.wghcwc.everyshowing.loadstyle.LoadingStyle;


/**
 * @author wghcwc
 * @date 20-1-5
 */
public class LoadingDefaultView extends LinearLayout {
    private ImageView ivBigLoading, ivSmallLoading;
    private SVCircleProgressBar circleProgressBar;
    private TextView tvMsg;
    private LoadingStyle mStyle;
    private LinearLayout background;

    public LoadingDefaultView(Context context, LoadingStyle style) {
        super(context);
        this.mStyle = style;
        initViews();
    }

    public LoadingDefaultView(Context context) {
        this(context, new BaseLoadingStyle());
    }


    private void initViews() {
        LayoutInflater.from(getContext()).inflate(R.layout.view_svprogressdefault, this, true);
        background = findViewById(R.id.background);
        background.setBackgroundResource(mStyle.getBackground());
        ivBigLoading = findViewById(R.id.ivBigLoading);
        ivBigLoading.setScaleType(mStyle.getScaleType());
        ivBigLoading.setPadding(mStyle.getPadding(), mStyle.getPadding(), mStyle.getPadding(), mStyle.getPadding());
        ivSmallLoading = findViewById(R.id.ivSmallLoading);
        circleProgressBar = findViewById(R.id.circleProgressBar);
        tvMsg = findViewById(R.id.tvMsg);
    }

    public void show() {
        clearAnimations();
        ivBigLoading.setImageResource(mStyle.getImage());
        ivBigLoading.setVisibility(View.VISIBLE);
        ivSmallLoading.setVisibility(View.GONE);
        circleProgressBar.setVisibility(View.GONE);
        tvMsg.setVisibility(View.GONE);
        ivBigLoading.startAnimation(mStyle.getAnimation());
    }

    public void showWithStatus(String string) {
        if (string == null) {
            show();
            return;
        }
        showBaseStatus(mStyle.getImage(), string);
        ivSmallLoading.startAnimation(mStyle.getAnimation());
    }

    public void showInfoWithStatus(String string) {
        showBaseStatus(mStyle.infoImage(), string);
    }

    public void showSuccessWithStatus(String string) {
        showBaseStatus(mStyle.successImage(), string);
    }

    public void showErrorWithStatus(String string) {
        showBaseStatus(mStyle.errorImage(), string);
    }

    public void showWithProgress(String string) {
        showProgress(string);
    }

    public SVCircleProgressBar getCircleProgressBar() {
        return circleProgressBar;
    }

    public void setText(String string) {
        tvMsg.setText(string);
    }

    public void showProgress(String string) {
        clearAnimations();
        tvMsg.setText(string);
        ivBigLoading.setVisibility(View.GONE);
        ivSmallLoading.setVisibility(View.GONE);
        circleProgressBar.setVisibility(View.VISIBLE);
        tvMsg.setVisibility(View.VISIBLE);
    }

    public void showBaseStatus(int res, String string) {
        clearAnimations();
        ivSmallLoading.setImageResource(res);
        tvMsg.setText(string);
        ivBigLoading.setVisibility(View.GONE);
        circleProgressBar.setVisibility(View.GONE);
        ivSmallLoading.setVisibility(View.VISIBLE);
        tvMsg.setVisibility(View.VISIBLE);
    }

    public void dismiss() {
        clearAnimations();
    }

    private void clearAnimations() {
        ivBigLoading.clearAnimation();
        ivSmallLoading.clearAnimation();
    }

}

File: everyshowing/src/main/java/com/wghcwc/everyshowing/view/LoadingDefaultView.kt
package com.wghcwc.everyshowing.view

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import com.wghcwc.everyshowing.R
import com.wghcwc.everyshowing.loadstyle.BaseLoadingStyle
import com.wghcwc.everyshowing.loadstyle.LoadingStyle
import kotlinx.android.synthetic.main.view_svprogressdefault.view.*

/**
 * @author wghcwc
 * @date 20-1-5
 */
class LoadingDefaultView @JvmOverloads constructor(context: Context?, private val mStyle: LoadingStyle = BaseLoadingStyle()) : LinearLayout(context) {
    val circleProgressBar: SVCircleProgressBar? = null

    private fun initViews() {
        LayoutInflater.from(context).inflate(R.layout.view_svprogressdefault, this, true)
        llParent.setBackgroundResource(mStyle.background)
        ivBigLoading.scaleType = mStyle.scaleType
        ivBigLoading.setPadding(mStyle.padding, mStyle.padding, mStyle.padding, mStyle.padding)
    }

    fun show() {
        clearAnimations()
        ivBigLoading.setImageResource(mStyle.image)
        ivBigLoading.visibility = View.VISIBLE
        ivSmallLoading.visibility = View.GONE
        circleProgressBar?.visibility = View.GONE
        tvMsg.visibility = View.GONE
        ivBigLoading.startAnimation(mStyle.animation)
    }

    fun showWithStatus(string: String?) {
        if (string == null) {
            show()
            return
        }
        showBaseStatus(mStyle.image, string)
        ivSmallLoading.startAnimation(mStyle.animation)
    }

    fun showInfoWithStatus(string: String?) {
        showBaseStatus(mStyle.infoImage, string)
    }

    fun showSuccessWithStatus(string: String?) {
        showBaseStatus(mStyle.successImage, string)
    }

    fun showErrorWithStatus(string: String?) {
        showBaseStatus(mStyle.errorImage, string)
    }

    fun showWithProgress(string: String?) {
        showProgress(string)
    }

    fun setText(string: String?) {
        tvMsg.text = string
    }

    fun showProgress(string: String?) {
        clearAnimations()
        tvMsg.text = string
        ivBigLoading.visibility = View.GONE
        ivSmallLoading.visibility = View.GONE
        circleProgressBar?.visibility = View.VISIBLE
        tvMsg!!.visibility = View.VISIBLE
    }

    fun showBaseStatus(res: Int, string: String?) {
        clearAnimations()
        ivSmallLoading.setImageResource(res)
        tvMsg.text = string
        ivBigLoading.visibility = View.GONE
        circleProgressBar?.visibility = View.GONE
        ivSmallLoading.visibility = View.VISIBLE
        tvMsg.visibility = View.VISIBLE
    }

    fun dismiss() {
        clearAnimations()
    }

    private fun clearAnimations() {
        ivBigLoading.clearAnimation()
        ivSmallLoading.clearAnimation()
    }

    init {
        initViews()
    }
}